src/ublox.cpp

```diff
@@ -90,40 +125,54 @@ void UBLOX::set_baudrate(const uint32_t baudrate)
   out_message_.CFG_PRT.outProtoMask = CFG_PRT_t::OUT_UBX | CFG_PRT_t::OUT_NMEA;
   out_message_.CFG_PRT.mode = CFG_PRT_t::CHARLEN_8BIT | CFG_PRT_t::PARITY_NONE | CFG_PRT_t::STOP_BITS_1;
   out_message_.CFG_PRT.flags = 0;
-  send_message(CLASS_CFG, CFG_PRT, out_message_, sizeof(CFG_PRT_t));
+  send_ubx_message(CLASS_CFG, CFG_PRT, out_message_, sizeof(CFG_PRT_t));
+  delayMicroseconds(10000);
```

Comment:
this is probably too long

Reply:
I changed it to wait for the TX buffer to empty rather than a set period of time.